### DISCUSSION TITLE
Should we encourage contributions to open source software as part of the Portfolio module?

### DISCUSSION THREAD

**Discussion by alismith-dev** [CodeYourFuture/curriculum#707] [Ideas]
Initially discussed in Tech Ed Syllabus team on 2024-02-13, [see minutes](https://curriculum.codeyourfuture.io/guides/contributing/minutes/#outreachy).

Originally inspired by [Outreachy](https://www.outreachy.org/), which is an organisation that arranges "internships" that contribute to open source software (OSS) projects for recently graduated bootcamp students & self-taught developers.

This led to a question of whether we could encourage contribution to OSS projects through the Portfolio module, which I'd like to open for discussion here.

**Comment by alismith-dev** [comment]
An obvious drawback: OSS projects have a reputation of being unfriendly to newcomers, leading to trainees feeling hurt & demotivated.

**Reply by alismith-dev** [reply]
We discussed the idea of CYF operated projects being a "safe" project for trainees to practice contributions.

@SallyMcGrath created https://github.com/CodeYourFuture/First-Timers-Project to provide guidance around this.

**Reply by FridIsar** [reply]
Interestingly this is also a discussion we're having in Barcelona:

Recently, GitHub Education has partnered with [OpenSauced](https://opensauced.pizza/students?mc_cid=3a6fd3d98f&mc_eid=67b48ff997), a social media for Open-Source contributions. Verified GitHub students can connect their OpenSauced profile to their LinkedIn profile for free (which showcases their contributions). 

GitHub Education has made an [Intro to Open-Source](https://education.github.com/experiences/intro_to_open_source) course with a "roadmap" (which seems more like a bunch of resources). 

@isha789 (graduate & volunteer) volunteered to do the whole course. Once she's done it, I'll ask her to compare it with Sally's course and will report our findings back to you!
